broker/fragment/store_azure.go

```diff
@@ -34,20 +35,27 @@ type AzureStoreConfig struct {
 	RewriterConfig
 }
 
-func (cfg *AzureStoreConfig) serviceUrl() string {
+func (cfg *azureStoreConfig) serviceUrl() string {
 	return fmt.Sprintf("https://%s.blob.core.windows.net", cfg.storageAccountName)
 }
 
-func (cfg *AzureStoreConfig) containerURL() string {
+func (cfg *azureStoreConfig) containerURL() string {
 	return fmt.Sprintf("%s/%s", cfg.serviceUrl(), cfg.containerName)
 }
 
+type udcAndExp struct {
+	udc *service.UserDelegationCredential
+	exp *time.Time
+}
+
 type azureBackend struct {
-	clients   map[string]pipeline.Pipeline
-	svcClient service.Client
-	clientMu  sync.Mutex
-	udc       *service.UserDelegationCredential
-	udcExp    *time.Time
+	// This is a cache of configured Pipelines for each tenant. These do not expire
+	clients map[string]pipeline.Pipeline
```

Comment:
nit: can these be called `pipelines` and `services` perhaps, here and elsewhere? Or `pipelines` and `clients` ?

It's hard to follow which is which.